### ISSUE TITLE
Slow response time to fast changes

### ISSUE THREAD

**Issue by RafigRzayev** [xioTechnologies/Fusion#26]
Hello, 

I am using this library without magnetometer in order to get pitch/roll angles. IMU Gyro/Accelerometer samples are taken 10 times per second. 

 The gain value used in `FusionAhrsInitialise() `is **0.5** as per recommendations in the code comments. 

Everything works fine, except that the response to angular changes is quite slow. For example, if I rotate the sensor 90 degrees instantaneously, I expect the output angles to change from 0 to 90 within short period of time. With this gain value, the change takes app. 4-5 seconds.  When I increase the gain to values above **3**, the response time improves greatly (to 1s) which is acceptable to the application, however the angles become pretty volatile and sensitive which is not that good. 

I wanted to ask if this behavior is normal? Why the recommended gain value results in such high delay in the output? Is there any other parameter that can be changed in order to speed up the the response? 

Current setup:
Sensor: icm42670p
Application sampling rate: 10Hz
Gyro range: 250_DPS
Accelerometer range: 2G

**Comment by xioTechnologies** [comment]
A sample rate of 10 Hz is far to low to measure a sensor rotated "90 degrees instantaneously".  I suggest you use a sample of 100 Hz or higher.

**Comment by RafigRzayev** [comment]
Thanks a lot for your help. I tried sampling rate of 100Hz, it doesn't change the behavior. To me it seems that there is no effect at all. Please see the results with both frequencies below:
![Capture](https://user-images.githubusercontent.com/58329539/156174604-83911772-309a-43ae-9537-9e655485ca43.JPG)

**Comment by RafigRzayev** [comment]
I solved it, there were two problems:
1) Instead of passing raw data of gyro/accel in lsb units I was passing them as converted into dps/g units. 
After fixing it the behavior didn't change. Then I increased the range of accelerometer and gyroscope from minimum settings to maximum, and this time everything worked great. In low settings, sensors reach their measurement cap and thus don't provide accurate input into the algorithm. 

Thank you for your help.
